Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -60,6 +61,9 @@ theorem respectsIso : MorphismProperty.RespectsIso @IsSeparated := by
   apply MorphismProperty.RespectsIso.diagonal
   exact IsClosedImmersion.respectsIso
 
+instance {X Y : Scheme} (f : X ⟶ Y) [IsSeparated f] : QuasiSeparated f where
```

Comment:
```suggestion
instance (priority := 900) {X Y : Scheme} (f : X ⟶ Y) [IsSeparated f] : QuasiSeparated f where
```

Reply:
Changed. I am seeing this in a lot of places: why do we need this or why is it advisable?

Reply:
Because lean finds type classes in reverse. It sees that the goal is `QuasiSeparated f` and it lists all instances with goals looking like that and tries it one by one. Usually `f` is some complex term whose structure gives instances (for example `QuasiSeparated f -> QuasiSeparated (pullback f g)`), and instances with generic goals (i.e. `f` is a variable) should be tried last.

Reply:
There's a note [lower instance priority] for this.